addons/calendar/tests/test_calendar.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import time
+
+from odoo.tests import common
+
+
+class TestCalendar(common.TransactionCase):
+
+    def setUp(self):
+        super(TestCalendar, self).setUp()
+
+        self.CalendarEvent = self.env['calendar.event']
+        #In Order to test calendar, I will first create One Simple Event with real data
+        self.calendar_event = self.CalendarEvent.create({
+            'privacy': 'private',
+            'start': '2011-04-30 16:00:00',
+            'stop': '2011-04-30 18:30:00',
+            'description': 'The Technical Presentation will cover following topics:\n* Creating OpenERP class\n* Views\n* Wizards\n* Workflows',
+            'duration': 2.5,
+            'location': 'OpenERP S.A.',
+            'name': 'Technical Presentation'})
+
+        #Now I will set recurrence for this event to occur monday and friday of week
+        self.data = {
+            'fr': 1,
+            'mo': 1,
+            'interval': 1,
+            'rrule_type': 'weekly',
+            'end_type': 'end_date',
+            'final_date': '2011-05-31 00:00:00',
+            'recurrency': True}
+
+    def test_calender_event(self):
+        self.calendar_event.write(self.data)
+
+        # In order to check that recurrent events are views successfully in calendar view, I will open calendar view of events|
+        self.calendar_event.fields_view_get(False, 'calendar')
+
+        #In order to check that recurrent events are views successfully in calendar view, I will search for one of the recurrent event and count the number of events
+        rec_events = self.CalendarEvent.with_context({'virtual_id': True}).search([('start', '>=', '2011-04-30 16:00:00'), ('start', '<=', '2011-05-31 00:00:00')])
+        self.assertEqual(len(rec_events), 9, 'Wrong number of events found')
+
+        #Now I move a virtual event, to see that a real event is well created and depending from the native recurrence
+        before = self.CalendarEvent.with_context({'virtual_id': False}).search([('start', '>=', '2011-04-30 16:00:00'), ('start', '<=', '2011-05-31 00:00:00')])
+
+        # We start by detach the event
+        newevent = rec_events[1]._detach_one_event()
+        newevent.with_context({'virtual_id': True}).write({'name': 'New Name', 'recurrency': True})
+        after = self.CalendarEvent.with_context({'virtual_id': False}).search([('start', '>=', '2011-04-30 16:00:00'), ('start', '<=', '2011-05-31 00:00:00')])
+        self.assertEqual(len(after), len(before)+1, 'Wrong number of events found, after to have moved a virtual event')
+        new_event = list(set(after)-set(before))[0]
```

Comment:
Do you need to convert it to set and list, its already recorsset, you can do subtraction directly